StyleCop.Analyzers/StyleCop.Analyzers.Test/LayoutRules/SA1500/SA1500UnitTests.Classes.cs

```diff
@@ -93,6 +125,8 @@ public class InvalidClass6
             };
 
             await this.VerifyCSharpDiagnosticAsync(testCode, expectedDiagnostics, CancellationToken.None).ConfigureAwait(false);
+            await this.VerifyCSharpDiagnosticAsync(fixedTestCode, EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
+            await this.VerifyCSharpFixAsync(testCode, fixedTestCode).ConfigureAwait(false);
```

Comment:
No, it does not. It verifies if the code fix doesn't introduce compiler diagnostics. It does not check if the proposed fixed code is clean. 

I found the pattern above in one of the other unit test sets and liked it so much that I started using it as well. Instead of adapting VerifyCSharpFixAsync to also test for invalid fixed code, I think we should wrap up the three verify statements above into a single method, something like VerifyCSharpDiagnosticAndFixAsync.